File: backend-dotnetcore/ApiModels/Actor.cs
using Microsoft.AspNetCore.Mvc.ModelBinding;
using Swashbuckle.AspNetCore.Annotations;

namespace RUG.WebEng.Movies.ApiModels;

public class Actor
{
    [BindNever]
    [SwaggerSchema(ReadOnly = true)]
    public int Id { get; init; }
    public string Name { get; set; }
    public List<MovieSummary> Movies { get; set; } = new();

    public static Actor FromDatabase(Models.Actor actor) =>
        new()
        {
            Id = actor.Id,
            Name = actor.Name,
            Movies = actor.Movies.Select(MovieSummary.FromDatabase).ToList()
        };
}

File: backend-express/api-models/actor.ts
import { MovieSummary } from "./movie-summary";
import { Actor as DbActor } from "models/actor";

export class Actor {
    id: number;
    name: string;
    movies: MovieSummary[] = [];

    public static fromDatabase(actor : DbActor) : Actor {
        return {
            id: actor.id,
            name: actor.name,
            movies: actor.movies?.map(MovieSummary.fromDatabase) ?? []
        };
    }
}